.github/workflows/cilium-cli-test.yaml

```diff
@@ -0,0 +1,160 @@
+name: Run cilium connectivity test against the latest Cilium release
+
+on:
+  pull_request: {}
+
+permissions: read-all
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number }}
+  cancel-in-progress: true
+
+env:
+  KIND_CONFIG: .github/kind-config-extra-nodes.yaml
+
+jobs:
+  check_changes:
+    name: Deduce required tests from code changes
+    runs-on: ubuntu-22.04
+    outputs:
+      tested: ${{ steps.tested-tree.outputs.src }}
+    steps:
+      - name: Check code changes
+        uses: dorny/paths-filter@de90cc6fb38fc0963ad72b210f1f284cd68cea36 # v3.0.2
+        id: tested-tree
+        with:
+          filters: |
+            src:
+              - 'cilium-cli/**'
+
+  cilium-connectivity-test:
+    needs: check_changes
+    if: ${{ needs.check_changes.outputs.tested == 'true' }}
+    runs-on: ubuntu-latest
+    name: Run cilium connectivity test against the latest Cilium release.
+    steps:
+      - name: Checkout target branch to access local actions
+        uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
+        with:
+          ref: ${{ github.base_ref || github.ref }}
+          persist-credentials: false
+
+      - name: Set Environment Variables
+        uses: ./.github/actions/set-env-variables
+
+      - name: Checkout
+        uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
+        with:
+          persist-credentials: false
+
+      - name: Create kind cluster
+        uses: helm/kind-action@99576bfa6ddf9a8e612d83b513da5a75875caced # v1.9.0
+        with:
+          version: ${{ env.KIND_VERSION }}
+          node_image: ${{ env.KIND_K8S_IMAGE }}
+          kubectl_version: ${{ env.KIND_K8S_VERSION }}
+          config: ${{ env.KIND_CONFIG }}
+          wait: 0 # The control-plane never becomes ready, since no CNI is present
+
+      - name: Install Cilium CLI
+        uses: cilium/cilium-cli@de740a64da1b12dc8488cf3af2474c60d52c349e # v0.16.3
+        with:
+          go-mod-directory: '.'
+
+      - name: Install Cilium
+        run: |
+          cilium install \
+            --nodes-without-cilium \
+            --wait=false \
+            --set bpf.monitorAggregation=none \
+            --set cni.chainingMode=portmap \
+            --set hubble.relay.enabled=true \
+            --set loadBalancer.l7.backend=envoy \
+            --set tls.secretsBackend=k8s \
+            --set prometheus.enabled=true \
+            ${{ env.CILIUM_INSTALL_EXTRA_FLAGS }}
```

Comment:
I assume this will install the latest Cilium release defined here: https://github.com/cilium/cilium-cli/blob/756ae5072a7281e2b6b33b10da3ab890cb2fa240/defaults/defaults.go#L105

I wonder whether we should explicitly define the version as part of this workflow in order to not be reliant on the Cilium version bump in the cilium/cilium-cli repo.

Reply:
yeah makes sense let me see what i can do

Reply:
💡 https://github.com/cilium/cilium/blob/a995f3bd90372006db54c983bea253ab3efdc8c7/.github/actions/set-env-variables/action.yml#L29